File: examples/dgemm.R
a <- matrix(1:16, ncol = 2048, nrow = 2048)
b <- matrix(1:16, ncol = 2048, nrow = 2048)
c <- matrix(1:16, ncol = 2048, nrow = 2048)

start <- proc.time()[3]
c <- a %*% b
end <- proc.time()[3]

end - start

g <- 2*2048*2048*2048/(end - start)
"Gflops"
g*1e-9

File: examples/dgemm.py
#!/usr/bin/python

import time
import numpy
from numpy.random import randn

def run_dgemm():

	N=3840*2
	
	print("dgemm %dx%d" % ( N, N ));

	A = randn(N,N).astype('float64')
	B = randn(N,N).astype('float64')

	start = time.time();
	ref = numpy.dot(A,B)
	end = time.time()
	
	timediff = (end -start) 
	gflops = ( 2*N*N*N) / timediff
	gflops *= 1e-9

	print("Time:\t\t%f" % timediff);
	print("GFlops:\t\t%f" % gflops);


if __name__ == "__main__":
	run_dgemm()

